tarogarden/seedling.go

```diff
@@ -124,6 +128,49 @@ func (c Seedling) validateFields() error {
 	return nil
 }
 
+// validateGroupKey attempts to validate that the non-zero group key provided
+// with a seedling is owned by the daemon and can be used with this seedling.
+func (c Seedling) validateGroupKey(groups []*asset.AssetGroup) (
+	*asset.AssetGroup, error) {
+
+	if len(groups) == 0 {
+		return nil, fmt.Errorf("group key %v not found",
+			c.GroupInfo.GroupKey.GroupPubKey.SerializeCompressed())
+	}
+
+	// Asset groups are sorted in insertion order, so the first group should
+	// include the information used for the original tweak.
+	seedlingGroup := groups[0]
```

Comment:
Can just pass in the first one instead. Also see the other comment about just making sure we get distinct values from the disk.